app/components/modals/PassphraseModal/PassphraseModal.js

```diff
@@ -11,6 +11,10 @@ class PassphraseModal extends React.Component {
     this.state = this.getInitialState();
   }
 
+  componentDidMount() {
+    document.addEventListener("keydown", this.onKeyPress, false);
```

Comment:
Don't use globals if at all possible. This will cause this event handler to be called whenever any key anywhere in the app is pressed after the first modal is shown, will cause a stack of handlers being added (one for every time a modal is shown), is likely to cause problems given it's never removed (and will be called even after this component has been unmounted) etc.

Could this handler be added only on the top-level component of the modal?

Reply:
A quick search shows that `PassphraseModal` is referenced in many different places - managing the handler in multiple contexts seems error prone. What if the handler stayed on the modal but was removed when the modal is closed (in `onCancelModal`)? That would address the stack problem and be more DRY. Let me know what you think or if I misunderstood anything.

Reply:
By "top level component" I meant the top level component _inside_ the PassphraseModal (the `<Modal ...` instance).

So `Modal` would have a new optional prop (`onKeyDown`) propagated down its own stack, or maybe the cancelation via ESC could be implemented directly in `Modal`, so that *all* modals could be closed with that.